### ISSUE TITLE
Migration to xDAI (APM + Directory)

### ISSUE THREAD

**Issue by dapplion** [dappnode/DAppNode#327]
DAppNode will migrate it's package manager + package directory contracts. This contracts track package versions in a decentralized way, and are consumed by all DAppNodes.

**Summary**

- [ ] Re-design and deploy package manager contracts. We will not deploy APM contracts as mainnet, since they don't fullfil all our needs
- [ ] Re-design and deploy directory contracts. Consider in the future to upgrade to a TCR
- [ ] Deploy (or consume) ENS to support above two points
- [ ] Move existing versions to xDAI. How to migrate packages not own by DAppNode, maybe add a migration UI?
- [ ] Add xDAI package
- [ ] Change DAPPMANAGER package consumer code to use xDAI
- [ ] Change remote repository logic to use xDAI nodes
- [ ] Deploy and mantain an xDAI node for remote repository consumers

**Comment by q9f** [comment]
Does this mean we will get an APM version that supports patch labels? :D

**Comment by dapplion** [comment]
> Does this mean we will get an APM version that supports patch labels? :D [aragon/apm#8](https://github.com/aragon/apm/issues/8)

Yes! We plan to just use arbitrary version texts. Since we don't use smart contract versioning there's no need to force strict semver numerals

**Comment by ligi** [comment]
Excited to see movement in the package manager area! Linking back to a discussion around this in the forum: https://forum.dappnode.io/t/dappnode-repositories-in-xdai/1184
